lib/block_view_lockups.go

```diff
@@ -1754,6 +2612,60 @@ func (bav *UtxoView) _connectCoinUnlock(
 	return totalInput, totalOutput, utxoOpsForTxn, nil
 }
 
+func CalculateVestedEarnings(
```

Comment:
Feels like you can consolidate this with CalculateLockupOverElapsedDuration, or maybe call it in this function to save doing math twice. Keep the sanity-check at the end if you do though!